protocol/interop-topology.md

```diff
@@ -0,0 +1,217 @@
+it may be prudent to switch the active Sequencer to one with a functional Supervisor.
+
+## Solution Side-Ideas
+
+Although they aren't strictly related to TX Flow or Redundancy, here are additional ideas to increase the stability
+of a network. These ideas won't be brought forward into the Solution Summary.
+
+### `op-supervisor` alternative backend
+
+We add a backend mode to `op-supervisor` that operates specifically by using dynamic calls to `eth_getLogs`
+to validate cross chain messages rather than its local index. This could be accomplished by adding
+new RPC endpoints that do this or could be done with runtime config. When `op-supervisor` runs in this
+mode, it is a "light mode" that only supports `supervisor_validateMessagesV2` and `supervisor_validateAccessList`
+(potentially a subset of their behavior). This would give us a form of "client diversity" with respect
+to validating cross chain messages. This is a low lift way to reduce the likelihood of a forged initiating
+message. A forged initiating message would be tricking the caller into believing that an initiating
+message exists when it actually doesn't, meaning that it could be possible for an invalid executing
+message to finalize.
+
+TODO: feedback to understand what capabilities are possible
+
+
+# Solution Summary
+
+We should establish `op-supervisor` checks of transactions at the following points:
+- On cloud ingress to `proxyd`
+- On ingress to all mempools
+- On regular interval on all mempools
+
+Additionally, the Sequencer should batch the calls 
+
+The production topology of an interop cluster checks the validity of cross chain transactions at:
+- cloud ingress (`proxyd`)
+- sentry node mempool ingress
+- sentry node mempool on interval
+- sequencer node mempool on interval
+
+The validity of cross chain transactions are not checked at:
+- sequencer node mempool ingress
+- block builder
+
+It is safe to not check at block building time because if the check passes at ingress
+time and passes again at block building time, it does not exhaustively cover all cases.
+It is possible that the remote reorg happens **after** the local block is sealed.
+In practice, it is far more likely to have an unsafe reorg on the remote chain that is not
+caught **before** the local block is sealed because the time between checking at the mempool
+and checking at block building is so small. Therefore we should not add a remote RPC request
+as part of the block building happy path, but we should still implement and benchmark it.
+
+TODO: include information on recommendation around standard mode/multi node/running multiple supervisors.
+Should include information on the value props of each so we can easily align on roadmap.
+
+# Alternatives Considered
+
+<!-- List out a short summary of each possible solution that was considered.
+Comparing the effort of each solution -->
+
+## Block Building
+
```

Comment:
The biggest uncertainty around doing the checks in the block builder is for `rbuilder`, when it runs in a TEE. Technically, its possible to block transactions with a firewall before they reach the block builder, so the transactions would need to be encrypted to the TEE itself. I also think that the chain would need to validate a TEE proof alongside batches, because I do not think there is a way to guarantee the private key for the batcher isn't in 2 locations - inside and outside of the TEE. See https://github.com/ethereum-optimism/design-docs/pull/229 for a design that would enable validating a TEE proof onchain

We should descope the concept of using a TEE completely until we are able to end to end understand the security model

Reply:
I would say further that the scope of this document applies to what OP Labs plans on doing with our infrastructure, and strictly speaking is a matter of policy. Since we don't run `rbuilder`, I don't see much to consider.

+1: an operator using a TEE will need to decide on Supervisor checks at times that make sense to them

Reply:
This does not mean that its not useful to check at the block builder level, we have that feature implemented in `op-geth`. We may ultimately delete that code or keep it around and use it as a fallback or explicitly only use it for local devnets. We need to ship the code and measure its performance to truly know the ideal architecture